domene/src/main/kotlin/no/nav/tiltakspenger/domene/personopplysninger/SakPersonopplysninger.kt

```diff
@@ -0,0 +1,100 @@
+        liste.filterIsInstance<PersonopplysningerSøker>().firstOrNull()
+
+    fun søkerMedIdent(ident: String): PersonopplysningerSøker? =
+        liste.filterIsInstance<PersonopplysningerSøker>().firstOrNull { it.ident == ident }
+
+    fun søker(): PersonopplysningerSøker =
+        liste.filterIsInstance<PersonopplysningerSøker>().first()
+
+    fun barnMedIdent(): List<PersonopplysningerBarnMedIdent> =
+        liste.filterIsInstance<PersonopplysningerBarnMedIdent>()
+
+    fun barnMedIdent(ident: String): PersonopplysningerBarnMedIdent? =
+        liste.filterIsInstance<PersonopplysningerBarnMedIdent>().firstOrNull { it.ident == ident }
+
+    fun barnUtenIdent(): List<PersonopplysningerBarnUtenIdent> =
+        liste.filterIsInstance<PersonopplysningerBarnUtenIdent>()
+
+    fun personerMedIdent(): List<PersonopplysningerMedIdent> =
+        liste.filterIsInstance<PersonopplysningerMedIdent>()
+
+    fun erTom(): Boolean = liste.isEmpty()
+
+    fun harTilgang(saksbehandler: Saksbehandler): Boolean {
+        if (liste.any { it.strengtFortrolig() }) return Rolle.STRENGT_FORTROLIG_ADRESSE in saksbehandler.roller
+        if (liste.any { it.fortrolig() }) return Rolle.FORTROLIG_ADRESSE in saksbehandler.roller
+        if (liste.any { it.skjermet() }) return Rolle.SKJERMING in saksbehandler.roller
+        return true
+    }
+
+    fun identerOgSkjerming(): Map<String, Boolean?> =
+        personerMedIdent().associate {
+            it.ident() to try {
+                it.avklartSkjerming()
+            } catch (e: IllegalStateException) {
+                null
+            }
+        }
+
+    fun setSkjermingForPerson(ident: String, erSkjermet: Boolean?): SakPersonopplysninger {
+        return SakPersonopplysninger(
+            liste.map { personopplysninger ->
+                when (personopplysninger) {
+                    is PersonopplysningerBarnUtenIdent -> personopplysninger
+                    is PersonopplysningerBarnMedIdent ->
+                        if (personopplysninger.ident == ident) {
+                            personopplysninger.copy(skjermet = erSkjermet)
+                        } else {
+                            personopplysninger
+                        }
+
+                    is PersonopplysningerSøker ->
+                        if (personopplysninger.ident == ident) {
+                            personopplysninger.copy(skjermet = erSkjermet)
+                        } else {
+                            personopplysninger
+                        }
+                }
+            },
+        )
+    }
+
+    fun medSkjermingFra(identOgSkjerming: Map<String, Boolean?>): SakPersonopplysninger =
+        identOgSkjerming
+            .toList()
+            .fold(this) { personopplysninger: SakPersonopplysninger, personMedIdent: Pair<String, Boolean?> ->
+                personopplysninger.setSkjermingForPerson(personMedIdent.first, personMedIdent.second)
+            }
```

Comment:
```suggestion
    fun medSkjermingFra(identerOgSkjerming: Map<String, Boolean?>): SakPersonopplysninger =
        apply {
            identerOgSkjerming.forEach { (ident, skjermet) ->
                settSkjermingForPerson(ident, skjermet)
            }
        }
```

Er ikke dette enklere?